### PR TITLE
Add constants in the Config for the Lifetime of KeyPackages

### LINKED ISSUE #164: Auto generate lifetime extension with KeyPackageBundle
According to the spec, both the capabilities extension and the lifetime extension are mandatory. Right now, only the capabilities extension is auto generated.
The default lifetime should be specified in the config of the group.

### PR DESCRIPTION
**Pull request by kkohbrok**
Fixes #164.

This PR introduces a new `Constants` field in the `Config`, which (for now) only includes some defaults for KeyPackage lifetimes. The PR also includes a small test, as well as some refactoring to make use of the constants.

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -11,6 +11,15 @@ use crate::extensions::ExtensionType;
 pub mod errors;
 pub(crate) use errors::ConfigError;
 
+/// This value is used as the default lifetime of `KeyPackage`s if no default
+/// lifetime is configured. The value is in seconds and amounts to 3 * 28 Days,
+/// i.e. about 3 months.
+const DEFAULT_KEY_PACKAGE_LIFETIME: u64 = 60 * 60 * 24 * 28 * 3; // in Seconds
+/// This value is sued as the default amount of time (in seconds) the lifetime
```
Comment (franziskuskiefer) [inline src/config/mod.rs:18]:
```suggestion
/// This value is used as the default amount of time (in seconds) the lifetime
```

File: src/config/mod.rs [hunk 2]
```diff
@@ -103,6 +130,17 @@ impl Config {
             None => Err(ConfigError::UnsupportedCiphersuite),
         }
     }
+
+    /// Get the default `KeyPackage` lifetime (in seconds).
+    pub fn default_key_package_lifetime() -> &'static u64 {
```
Comment (franziskuskiefer) [inline src/config/mod.rs:135]: For primitives we should just return the value.

File: src/extensions/life_time_extension.rs [hunk 3]
```diff
@@ -20,29 +20,29 @@
...
 
 impl LifetimeExtension {
-    /// Create a new lifetime extensions with lifetime `t`.
+    /// Create a new lifetime extensions with lifetime `t` (in seconds).
     /// Note that the lifetime is extended 1h into the past to adapt to skewed
     /// clocks.
     pub fn new(t: u64) -> Self {
-        // TODO: #85 make the margin configurable.
-        const LIFETIME_MARGIN: u64 = 60 * 60;
+        let lifetime_margin: u64 = Config::key_package_lifetime_margin().clone();
```
Comment (franziskuskiefer) [inline src/extensions/life_time_extension.rs:40]: This should go away when you change the API and drop the clone. `u64` is copyable.

File: tests/test_config.rs [hunk 4]
```diff
@@ -57,3 +57,12 @@ fn default_ciphersuites() {
         supported_ciphersuites
     );
 }
+
+#[test]
+fn default_constants() {
+    // Make sure the supported ciphersuites are what we expect them to be.
+    let default_key_package_lifetime = Config::default_key_package_lifetime().clone();
```
Comment (franziskuskiefer) [inline tests/test_config.rs:64]: No clones needed.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks. lgtm with a couple nits.

File: src/config/mod.rs (hunk 2 above)
Comment (kkohbrok) [reply src/config/mod.rs:135]: I was wondering about that. Will do.

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Looks good except the clippy warning!

**Review by raphaelrobert — APPROVED** [review]

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**